libraries/material-networking/src/material-networking/TextureCache.cpp

```diff
@@ -252,6 +254,10 @@ NetworkTexturePointer TextureCache::getTexture(const QUrl& url, image::TextureUs
     if (url.scheme() == RESOURCE_SCHEME) {
         return getResourceTexture(url);
     }
+    QString decodedURL = url.toDisplayString(QUrl::FullyDecoded);
```

Comment:
I've integrated the requested changes and (I think) fixed the infinite loop issues, but after merging master, referencing textures by ID (this line specifically) no longer works.  the issue is that entity IDs are treated as texture urls, so they are encoded.  an entity ID like "{2cc07c81-d04f-4d80-b29a-99601ad92769}" becomes "%7B2cc07c81-d04f-4d80-b29a-99601ad92769%7D".  this line is supposed to bring back the curly brackets...but it seems like it no longer does?  I'm not sure if this is a regression due to the Qt upgrade or what.  @ctrlaltdavid do you have any ideas?  I wonder if this affects other areas of the codebase

Reply:
I assume that the leading `%7B` and trailing `%7D` are actually there. ...
Perhaps `QUrl::PrettyDecoded` would work?
Or do QString replaces on `%7B` and `%7D`.
Do you have a JSON with example entity etc. for easy dev testing?

Reply:
I worked around this with `QUrl::fromPercentEncoding` 👍 

a test is to make a web entity and an image entity, and set the image entity's image url to the web entities ID